server/protocols/webadmin/webadmin-jmap/src/main/java/org/apache/james/webadmin/data/jmap/RecomputeUserFastViewProjectionItemsTask.java

```diff
@@ -118,15 +119,15 @@ public String getUsername() {
     }
 
     @Override
-    public Result run() {
-        corrector.correctUsersProjectionItems(progress, username)
+    public Publisher<Result> runReactive() {
+        return corrector.correctUsersProjectionItems(progress, username)
             .subscribeOn(Schedulers.elastic())
-            .block();
-
-        if (progress.failed()) {
-            return Result.PARTIAL;
-        }
-        return Result.COMPLETED;
+            .map(any -> {
+                if (progress.failed()) {
```

Comment:
I do not get what happen here, is it blocking?

Reply:
This code relies on a side effect rather than on a map + reduce.

I don't intend to change that here...

Reply:
This is fixed here: https://github.com/linagora/james-project/pull/3377/commits/1ee8a35da336564efbf1aa15062d75ff146b7fde